File: src/quicksort.cpp
#include<iostream>
#include<vector>

using namespace std;

void quickSort(vector<int> &a, int left, int right) {
	int i, j, temp;
	i = left;
	j = right;
	temp = a[left];
	if (left > right) {
		return ;
		}
	//temp = a[left];
	while (i!= j) {
          while (a[j] >= temp && i < j) { 
		  j--;
		  }
	  while (a[i] <= temp && i < j) { 
		  i++;
		  }
	  if (i < j) { 
		  int t;
		  t = a[i];
		  a[i] = a[j];
		  a[j] = t;
		  }
	//int size = a.size();
	//for(int i = 0; i < size;i++){
	 //cout << a[i] << " ";
          //        }
	}
	//cout << "111"<< endl;
	a[left] = a[i];
	a[i] = temp;
	int size = a.size();
	for(int k = 0; k < size; k++) {
		cout << a[k] << " ";
		}
	cout << "\n";
	quickSort(a, left, i-1);
	quickSort(a, i+1, right);
}

int main() {
	int a [6] = {4,5,1,3,2,2};
	vector<int> b(a,a+6);
	quickSort(b,0,5);
	int size = b.size();
	for(int i = 0; i < size;i++){
		cout << b[i] << " ";
		}
	}

File: python/quicksort.py
#-*- coding=utf-8 -*-

def quickSort(arr, left, right):
	if left >= right:
			return
	#mid = left + (right - left) // 2
	pivot = arr[left]
	i = left
	j = right
	while i != j:
			while arr[j] > pivot and i < j:
					j -= 1
			while arr[i] <= pivot and i < j:
					i += 1

			if (i < j):
					arr[j], arr[i] = arr[i], arr[j]
	arr[left], arr[i] = arr[i], arr[left]
	print(arr)
	quickSort(arr, 0, i-1)
	quickSort(arr, i+1, right)

def quickSort1(arr):
	if len(arr) < 2:
			return arr
	pivot = arr[0]
	resleft = [k for k in arr[1:] if k <= pivot]
	resright = [k for k in arr[1:] if k > pivot]
	return quickSort1(resleft) + [pivot] + quickSort1(resright)

arr = [7,4,11,4,3,9,1]

arr = quickSort1(arr)

print(arr)
	
	




